### PR TITLE
term interface 2

### PR DESCRIPTION
**Pull request by jverzani**
try to get updated TermInterface working

### CODE REVIEW

File: src/lambdify.jl [hunk 1]
```diff
@@ -105,9 +105,13 @@ end
...
 
-function _similarterm(t::SymbolicObject, f, args, symtype=nothing;
-                      metadata=nothing, exprhead=:call)
-    f(args...) # default
+function _similarterm(T::Type{<:SymbolicObject}, head::Symbol, args, metadata)
+    if head == :call
+        f, as... = args
+        return f(as...)
+    else
+        @show :head
```
Comment (ChrisRackauckas) [inline src/lambdify.jl:113]: ?

Comment (jverzani) [reply src/lambdify.jl]: Yes, this was a mess. I think I better understand what to do here now after reading what Symbolics did (and not what is being done for Expr in TermInterface.

**Comment by 0x0f0f0f** [comment]
failure on nightly CI due to an ambiguous overriding of `write`. maybe unrelated

**Comment by jverzani** [comment]
Thanks. That seems unrelated. Let me merge and tag this and see if it solve the issue. If not, I'll disable the extension I have in SymPyPythonCall and see if that solves it.

**Pull request merged**